### PR TITLE
Adds Trick Revolver

### PR DESCRIPTION
**Pull request by Fox-McCloud**
Inspired by Goon's trick revolver.

Adds the trick revolver, a clown only traitor item that fires backwards and instakills whoever tries to use it. Teach those pesky validhunters a lesson while getting a good laugh, to boot! Available for only 1 telecrystal.

Refactors Russian revolver code a bit to accommodate for the new behavior.

Behavior changes to Russian Revolver:
- Clicking the revolver anywhere will attempt to cause it to fire (and it will always hit whoever fires it, if it goes off)
- If the user doesn't have a head, it will instead hit their chest
- Damage increased from 200 to 300 in addition to dealing 560 bleed damage

Other
- Admins can now summon guns with only revolvers
- There's a more evil version that has a 50% chance to give you a real revolver or a trick revolver

:cl: Fox McCloud
add: Adds trick revolver to the syndicate uplink for the clown. Teach those pesky validhunters a lesson!
add: Adds the ability for admins to start a revolver only summon gun version; there's also a suicidal version that makes half the guns summoned real and the other half trick revolvers
/:cl:

### CODE REVIEW

**Comment by Spacemanspark** [comment]
Didn't we have a PR to add this item in the past, which was opposed?
Either way, not really sure how I feel about this.

**Comment by Fox-McCloud** [comment]
That was the chameleon bomb you're thinking of.

**Comment by FreeStylaLT** [comment]
RIP 😢

**Comment by FreeStylaLT** [comment]
oh okay

**Comment by TDSSS** [comment]
You probably want to exclude those from surplus crates, or many unlucky traitors will blow their own brains out.

**Comment by taukausanake** [comment]
I could have sworn there was a gun too that shot the user instead. Can't find anything. Maybe it was the bomb :man_shrugging:

**Comment by Twinmold93** [comment]
In regards to identifying the weapon, is there anyone way to do so if you spend long enough trying to figure it out or as an antag_desc only be able to identify its real purpose?

**Comment by Ralta** [comment]
Would this be stopped by holding a shield in your other hand? 

I assume it _shouldn't_ be, but as a clumsy clown, I've shot myself before and had the bullet get stopped by the shield I had in my other hand (on a low chance).

**Comment by Fox-McCloud** [comment]
Nope. You're going to die if it goes off.

File: code/game/objects/items/toys.dm [hunk 1]
```diff
@@ -1372,54 +1372,79 @@ obj/item/toy/cards/deck/syndicate/black
...
+		return FALSE
+	if(bullets_left == 1)
+		bullets_left = 0
+		var/zone = "head"
+		if(!(user.has_organ(zone))) // If they somehow don't have a head.
+			zone = "chest"
+		playsound(src, 'sound/weapons/gunshots/gunshot_strong.ogg', 50, 1)
+		user.visible_message("<span class='danger'>[src] goes off!</span>")
+		post_shot(user)
+		user.apply_damage(300, BRUTE, zone, sharp = TRUE, used_weapon = "Self-inflicted gunshot wound to the [zone].")
```
Comment (KasparoVy) [inline code/game/objects/items/toys.dm]: Can't block this with no shield, fellow reviewers

**Comment by Ralta** [comment]
Nice one, thank you both for confirming! :-)

File: code/game/objects/items/toys.dm [hunk 2]
```diff
@@ -1372,54 +1372,79 @@ obj/item/toy/cards/deck/syndicate/black
...
+		user.bleed(BLOOD_VOLUME_NORMAL)
```
Comment (Fox-McCloud) [inline code/game/objects/items/toys.dm]: HE SHOT OUT ALL MY BLOOD.

**Pull request merged**